front/src/modules/pathfinding/components/Pathfinding/PathfindingV2.tsx

```diff
@@ -300,19 +300,35 @@ const Pathfinding = ({ pathProperties, setPathProperties }: PathfindingProps) =>
               await postPathProperties(pathPropertiesParams).unwrap();
 
             if (electrifications && geometry && operational_points) {
-              const pathStepsWihPosition = compact(pathSteps).map((step, i) => ({
-                ...step,
-                positionOnPath: pathfindingResult.path_items_positions[i],
-              }));
-              dispatch(updatePathSteps(pathStepsWihPosition));
-
               const suggestedOperationalPoints: SuggestedOP[] = formatSuggestedOperationalPoints(
                 operational_points,
                 geometry,
                 pathfindingResult.length
               );
 
-              const allVias = upsertViasInOPs(suggestedOperationalPoints, pathStepsWihPosition);
+              // We update existing pathsteps with coordinates, positionOnPath and kp corresponding to the new pathfinding result
+              const updatedPathSteps: PathStep[] = compact(pathSteps).map((step, i) => {
+                const correspondingOp = suggestedOperationalPoints.find(
+                  (suggestedOp) =>
+                    step &&
+                    'uic' in step &&
+                    suggestedOp.uic === step.uic &&
+                    suggestedOp.ch === step.ch &&
+                    suggestedOp.kp === step.kp
+                );
+
+                return correspondingOp
+                  ? {
+                      ...step,
+                      kp: correspondingOp.kp,
```

Comment:
This one shouldn't be necessary since we check `suggestedOp.kp === step.kp` above I think? (Alternatively, drop the `kp` check above for consistency with `TypeAndPathV2`?)

Reply:
I'm not sure why a user would use 2 operational points with same uic and ch but with different kps (it can happen because of imports inconsistencies) but I think we need to keep the condition. But you're right the l.323 can be removed.

Reply:
> I'm not sure why a user would use 2 operational points with same uic and ch but with different kps (it can happen because of imports inconsistencies) but I think we need to keep the condition.

If the DB has the "Gare de Lyon" operational point on two tracks, each track will get its own OP sharing the same uic and ch right? In that case the kp may be different for each track's OP. It seems like filtering on kp would be incorrect.

> Maybe we should keep the kp l.323 in case the kp changed between two pathfinding calculation but remove the suggestedOp.kp === step.kp, what do you think ?

I think that would be more consistent with the logic in `TypeAndPathV2`.

Reply:
The last commit drops the kp filtering, so looks good!